docs/topics/multiplatform/multiplatform-compatibility-guide.md

```diff
@@ -490,3 +490,200 @@ projects.
 **When do the changes take effect?**
 
 In Kotlin 1.9.0, a deprecation warning is introduced when the `android` name is used in Kotlin Multiplatform projects.
+
+<anchor name="declaring-multiple-targets"></anchor>
+## Declaring multiple targets of the same kind
+
+**What's changed?**
+
+We discourage declaring multiple targets of the same kind in a single Gradle project.
+
+Example:
+```kotlin
+kotlin {
+    jvm()
+    jvm("jvmKtor")
+    jvm("jvmOkio") // is not recommended and will produce deprecation warning
+}
+```
+
+**What's the best practice now?**
+
+There are two ways of how to avoid multiple target declaration:
+
+1. Creating different compilations
+
+    If you need to compile some sources with different parameters such as dependencies or compiler options for example
+    for functional tests. Then instead of declaring multiple targets you can create multiple compilations in one target.
+ 
+   <tabs group="build-script">
+   <tab title="Before" group-key="before">
+
+    ```kotlin
+   // Project ':myLibrary:integrationTests'
+   kotlin {
+      jvm("jvmKtor") // !!!
```

Comment:
I'd recommend to avoid using JVM-shared source sets without the specific the need. I.e., let's add `linuxX64` or something here.